File: MediaLendingService.Server/Dto/BookDto.cs
namespace MediaLendingService.Server.Dto;

public record BookDto(
    int Id,
    string Title,
    string Author,
    string Description,
    Uri CoverImage,
    string Publisher,
    DateOnly PublicationDate,
    LiteraryCategoryDto Category,
    string Isbn,
    int PageCount,
    bool IsCheckedOut = false
);

File: medialendingservice.client/src/app/models/books/book.dto.ts
import { LiteraryCategoryDto } from "./literary-category.dto";
import { DateOnlyDto } from "./date-only.dto";

export interface BookDto {
  id: number;
  title?: string;
  author?: string;
  description?: string;
  coverImage?: string;
  publisher?: string;
  publicationDate?: DateOnlyDto;
  category?: LiteraryCategoryDto;
  isbn?: string;
  pageCount: number;
  isCheckedOut: boolean;
}
